[19.0][MIG] partner_phone_secondary

bosd opened the pull request:
build upon #2024

ivantodorovich committed:
[ADD] partner_phone_secondary

(cherry picked from commit e0ccf142a97dcb708009c34d7a03100c2588f3cc)

oca-travis committed:
[UPD] Update partner_phone_secondary.pot

(cherry picked from commit f4bf3911e6acd3e5464f03f4563c41831f4aa3d6)

OCA-git-bot committed:
[ADD] icon.png

(cherry picked from commit 6718e10fa9a87742f6242626a7a5f7b58bbe87fb)

OCA-git-bot committed:
[UPD] README.rst

(cherry picked from commit be383af141e2b1c2384de9ecc7437cc5b96986cf)

primes2h committed:
Added translation using Weblate (Italian)

(cherry picked from commit 146e22905db7a6e63e1d9229bcf755ea0ae77b1d)

simonstromb committed:
Added translation using Weblate (Swedish)

(cherry picked from commit 08d8b03e1ffabac77987f65be276245c486ac7eb)

pere-aquarian committed:
Added translation using Weblate (Catalan)

(cherry picked from commit 1dbdadb465b4e891a8c947839db2e49ce457bbf7)

OCA-git-bot committed:
[UPD] README.rst

(cherry picked from commit 6e697aa5ccec66e63576e8d156d739bc1403427e)

Ivorra78 committed:
Added translation using Weblate (Spanish)

(cherry picked from commit 013a458af722d597681a758807249e1f88a2d1e8)

paradoxxxzero committed:
[IMP] partner_phone_secondary: pre-commit execution

(cherry picked from commit 2c3f067730dd27f864876bc3304ec271724985db)

paradoxxxzero committed:
[MIG] partner_phone_secondary: Migration to 18.0

(cherry picked from commit 3552fed36648751bff7d0db9524c357f9865124f)

bosd committed:
[MIG]: partner_phone_secondary: Migrate to 19.0

[IMP] partner_phone_secondary: improve coverage on phone_secondary.py

[IMP] partner_phone_secondary: Improve view

bosd commented:
/ocabot migration partner_phone_secondary

JordiMForgeFlow commented:
@bosd why removing the maintainer?

JordiMForgeFlow commented:
@bosd the field in the form view of the partner should be added like this: https://github.com/OCA/partner-contact/blob/19.0/partner_mobile/views/res_partner.xml#L22

AndreuOForgeFlow commented on partner_phone_secondary/views/res_partner.xml:
```diff
@@ -0,0 +1,17 @@
+        <field name="arch" type="xml">
+            <field name="phone" position="after">
+                <field name="phone2" widget="phone" />
+            </field>
```
```suggestion
            <xpath expr="//div[field[@name='phone']]" position="after">
                <div class="d-flex align-items-baseline w-md-50">
                    <i class="fa fa-fw me-1 fa-phone text-primary" title="Phone"/>
                    <field name="phone2" class="w-100" widget="phone" placeholder="Phone 2"/>
                </div>
            </xpath>
```
Since Odoo 19.0 layout has changed, I’d suggest inserting the Phone 2 container directly under the main phone field. Otherwise, the field is not displayed. I’ve verified this using this PR’s runbot.

AndreuOForgeFlow reviewed (changes requested):
Some changes 👍

JordiMForgeFlow commented:
@bosd will you let us know if you plan on fixing the PR soon?

bosd commented:
> @bosd why removing the maintainer?

It was an oopsie! Fixed now

bosd commented:
> @bosd will you let us know if you plan on fixing the PR soon?

@JordiMForgeFlow Fixed! Can you please review.

bosd commented:
@dranyel-bosd Can you pleease review?

AndreuOForgeFlow reviewed (approved):
LGTM! 💯

AEstLo reviewed (approved):
LGTM

luisDIXMIT reviewed (approved):
Code reviewed and tested on Runboat, LGTM! Maybe the three commits can be squashed into the migration commit, as they are derived from the migration process.

NICO-SOLUTIONS commented:
Beginning from the line with patch(...) in test_02_onchange_phone2_validation, this part isn’t covered in Codecov because _onchange_phone2_validation does nothing when phone_validation isn’t installed (the _phone_format attribute only exists if the addon is present).

Could we improve coverage by forcing is_phone_validation_functional = True, installing phone_validation and phonenumbers in CI, or adding a small isolated test that directly exercises the patch?

Is it feasible to cover this mocked part in Codecov, or is it inherently limited by the addon dependency?

bosd commented:
> Code reviewed and tested on Runboat, LGTM! Maybe the three commits can be squashed into the migration commit, as they are derived from the migration process.

Done

BhaveshHeliconia reviewed (approved):
Functional review LGTM!

luisDIXMIT commented:
It seems to be ready to merge.

bosd commented:
> It seems to be ready to merge.

Do you have the super powers?

etobella commented:
https://github.com/OCA/partner-contact/pull/2363#pullrequestreview-4448157663 :pray:

Commit:
[IMP] partner_phone_secondary: add field in contact popup and simplified view

Forward-port of #2363 to 19.0: add phone2 to the simplified/quick-create
partner form and to the contact (child_ids) popup, and align the main form
inheritance with 18.0.

bosd commented:
@etobella forward-ported #2363 in de9f72b0e: phone2 is now added to the simplified/quick-create partner form (`view_partner_simple_form`) and to the contact (`child_ids`) popup, and the main form inheritance is aligned with 18.0. 🙏

bosd commented:
Hoping that this 6 month old pr, with 6 approving reviews is good enough to merge :grin: 
@etobella  Can you push this one over the  :checkered_flag: line?

etobella reviewed (approved):
/ocabot merge nobump

that was the idea 😁
